test/unit/adapters/http.js

```diff
@@ -27,8 +27,29 @@ describe('supports http with nodejs', function () {
     }
   });
 
-  it('should respect the timeout property', function (done) {
+  it('should detect connect timeout', function (done) {
+    var success = false, failure = false;
+    var error;
+
+    axios.get('http://10.255.255.1/', {
```

Comment:
Is this ip have special meaning? Can we use a non-existed host or something else?

Reply:
Yes. It's a one of well-known private address, which can be used for non-routable address.

In general, Dropping TCP SYN packet is common way to simulate connect timeout artificially.
Using `http://google.com:81` as request URL will result same effect because Google's Firewall drops TCP SYN packet, but i thought using Google's URL is not reliable because Google can change their firewall policy in someday, which is unpredictable. So that's why i use non-routable address as request url.

I'll add some comments to test code for explanation.